bundle-workflow/python/lib/build_manifest.py

```diff
@@ -20,17 +20,25 @@ def __post_init__(self):
 @dataclass
 class Build:
     data: str
+    id: str = None
     name: str = None
     version: str = None
     dict: dict = None
+    location: str = None
 
     def __post_init__(self):
+        self.id = self.data['build-id'] if 'build-id' in self.data else uuid.uuid4().hex
         self.name = self.data['name']
         self.version = self.data['version']
         self.__set_dict()
 
+    def set_location(self, location):
```

Comment:
Now you can set `.location = ` and call `set_location`, the first one won't have the desired side-effect. I think you want to copy the value of location into the `dict` elsewhere